src/arrangement/core.cljc

```diff
@@ -78,10 +90,11 @@
         (compare-seqs rank a b)
 
         :else
-        (let [class-diff (compare (.getName (class a))
-                                  (.getName (class b)))]
+        (let [class-diff (compare (type-name a)
+                                  (type-name b))]
           (if (zero? class-diff)
-            (if (instance? java.lang.Comparable a)
-              (compare a b)
-              (compare (str a) (str b)))
```

Comment:
Hmm. I think there are decent arguments to be made in either case. Let's go with `str` for now and we can keep an eye out for potential cases where this creates problems.